wdl/transforms/new-base/src/main/scala/wdl/transforms/base/ast2wdlom/AstToWorkflowDefinitionElement.scala

```diff
@@ -26,26 +27,65 @@ object AstToWorkflowDefinitionElement {
+
+    val intermediateValueDeclarationStdoutCheck = checkStdIntermediates(bodyElements.filterByType[IntermediateValueDeclarationElement], StdoutElement, "stdout")
+    val intermediateValueDeclarationStderrCheck: ErrorOr[Option[String]] = checkStdIntermediates(bodyElements.filterByType[IntermediateValueDeclarationElement], StderrElement, "stderr")
+
+    val outputsSectionValidation: ErrorOr[Option[OutputsSectionElement]] = for {
+      outputValidateElement <- validateSize(bodyElements.filterByType[OutputsSectionElement], "outputs", 1): ErrorOr[Option[OutputsSectionElement]]
+      _ <- checkIfStdOutputExists(outputValidateElement, StdoutElement, "stdout")
+      _ <- checkIfStdOutputExists(outputValidateElement, StderrElement, "stderr")
+    } yield outputValidateElement
 
     val graphSections: Vector[WorkflowGraphElement] = bodyElements.filterByType[WorkflowGraphElement]
 
     val metaSectionValidation: ErrorOr[Option[MetaSectionElement]] = validateSize(bodyElements.filterByType[MetaSectionElement], "meta", 1)
     val parameterMetaSectionValidation: ErrorOr[Option[ParameterMetaSectionElement]] = validateSize(bodyElements.filterByType[ParameterMetaSectionElement], "parameterMeta", 1)
 
-    (inputsSectionValidation, outputsSectionValidation, metaSectionValidation, parameterMetaSectionValidation) mapN {
-      (validInputs, validOutputs, meta, parameterMeta) =>
+    (inputsSectionValidation, outputsSectionValidation, metaSectionValidation, parameterMetaSectionValidation, intermediateValueDeclarationStdoutCheck, intermediateValueDeclarationStderrCheck) mapN {
+      (validInputs, validOutputs, meta, parameterMeta, _, _) =>
       WorkflowDefinitionElement(name, validInputs, graphSections.toSet, validOutputs, meta, parameterMeta, sourceLocation)
     }
   }
 
+  def checkIfStdInputExist(inputSection: Option[InputsSectionElement], expressionType: FunctionCallElement, expressionName: String): ErrorOr[Option[String]] = {
+    inputSection match {
+      case Some(section) =>
+        if (section.inputDeclarations.flatMap(_.expression).exists(_.isInstanceOf[expressionType.type])) {
+          s"Workflow cannot have $expressionName expression in input section at workflow-level.".invalidNel
+        } else None.validNel
+      case None => None.validNel
+    }
+  }
+
+  def checkIfStdOutputExists[A](outputSection: Option[OutputsSectionElement], expressionType: FunctionCallElement, expressionName: String): ErrorOr[Option[String]] = {
+    outputSection match {
+      case Some(section) =>
+        if (section.outputs.map(_.expression).exists(_.isInstanceOf[expressionType.type])) {
+          s"Workflow cannot have $expressionName expression in output section at workflow-level.".invalidNel
+        } else None.validNel
+      case None => None.validNel
+    }
+  }
+
+  def checkStdIntermediates[A](intermediate: Vector[IntermediateValueDeclarationElement], expressionType: FunctionCallElement, expressionName: String): ErrorOr[Option[String]] = {
```

Comment:
Going one step further, these three functions are similar enough that I think you could probably (if so inclined) combine them into a single function like

```
def checkForExpressionTypes[A](
    elementsToCheck: Iterable[A],
    expressionFinder: A => Iterable[ExpressionElement],
    expressionTypesToFind: Iterable[FunctionCallElement], 
    expressionName: String) = {
  ???
}
```

Reply:
I chose to have them all separate in case another noob like me comes along and is working on this, but implementing it this way would be cleaner I suppose. I'm torn between the two!